tests/testthat/test-methods.R

```diff
@@ -170,30 +170,30 @@ local({
   lints <- lint(text = "a", linters = test_linter())
   lint <- lints[[1L]]
 
-  wrapped_strings <- c(
-    "[test_linter]\n    The\n    quick\n    brown\n    fox\n    jumps\n    over\n    the\n    lazy\n    dog.",
-    "[test_linter]\n    The quick brown\n    fox jumps over\n    the lazy dog.",
-    "[test_linter] The\n    quick brown fox jumps over the lazy\n    dog.",
-    "[test_linter] The quick brown fox jumps over the lazy dog."
-  )
+  widths <- c(10L, 20L, 40L, 80L)
+  test_names <- paste0(": width = ", widths)
 
   patrick::with_parameters_test_that(
-    "format.lint, format.lints, print.lint, print.lints support optional message wrapping",
+    "print.lint, print.lints support optional message wrapping",
     {
-      expect_match(format(lint, width = width), wrapped_string, fixed = TRUE)
-      expect_match(format(lints, width = width), wrapped_string, fixed = TRUE)
-      expect_output(print(lint, width = width), wrapped_string, fixed = TRUE)
-      expect_output(print(lints, width = width), wrapped_string, fixed = TRUE)
+      withr::with_options(c(lintr.format_width = width), {
+        expect_snapshot(print(lint))
+        expect_snapshot(print(lints))
+      })
+    },
+    .test_name = test_names,
+    width = widths
+  )
 
+  patrick::with_parameters_test_that(
+    "format.lint, format.lints support optional message wrapping",
+    {
       withr::with_options(c(lintr.format_width = width), {
-        expect_match(format(lint), wrapped_string, fixed = TRUE)
-        expect_match(format(lints), wrapped_string, fixed = TRUE)
-        expect_output(print(lint), wrapped_string, fixed = TRUE)
-        expect_output(print(lints), wrapped_string, fixed = TRUE)
+        expect_snapshot(format(lint))
```

Comment:
Ideally, we should be doing `expect_snapshot(cat(format(lint)))` here, but that leads to an error.

IINM, this is related to an issue reported in https://github.com/r-lib/testthat/issues/1646

Reply:
WDYT, @MichaelChirico, @AshesITR ?

Reply:
I'm still not sure about using snapshot testing for the `format()` output. I lean towards testing the string output directly & leaving the snapshot for `print()`. I'll let @AshesITR weigh in.

Reply:
I agree with @MichaelChirico here. `print()` looks good as a snapshot test, though.

Reply:
Okay, I have restored tests for `format()` and using snapshot tests only for `print()`.